### PR TITLE
[feature-wip](multi-catalog) Support runtime filter for file scan node

### PR DESCRIPTION
**Pull request by morningman**
# Proposed changes

Issue Number: close #xxx

## Problem Summary:

1. Support runtime filter for file scan node
2. Refactor some profile of file scan node

## Checklist(Required)

1. Does it affect the original behavior: (No)
4. Has unit tests been added: (No)
5. Has document been added or modified: (No Need)
6. Does it need to update dependencies: (No)
7. Are there any changes that cannot be rolled back: (No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/vec/exec/file_scan_node.cpp [hunk 1]
```diff
@@ -90,18 +115,102 @@ Status FileScanNode::open(RuntimeState* state) {
...
 }
 
+Status FileScanNode::_acquire_and_build_runtime_filter(RuntimeState* state) {
+    // acquire runtime filter
+    _runtime_filter_ctxs.resize(_runtime_filter_descs.size());
+    for (size_t i = 0; i < _runtime_filter_descs.size(); ++i) {
+        auto& filter_desc = _runtime_filter_descs[i];
+        IRuntimeFilter* runtime_filter = nullptr;
+        state->runtime_filter_mgr()->get_consume_filter(filter_desc.filter_id, &runtime_filter);
+        DCHECK(runtime_filter != nullptr);
```
Comment (Jibing-Li) [inline be/src/vec/exec/file_scan_node.cpp:132]: What will happen if DCHECK(false)?

Comment (Jibing-Li) [inline be/src/vec/exec/file_scan_node.cpp:132]: Will DCHECK(false) throw an exception? I didn't find its definition.

File: fe/fe-core/src/main/java/org/apache/doris/planner/RuntimeFilterGenerator.java [hunk 2]
```diff
@@ -321,7 +322,7 @@ private void finalizeRuntimeFilter(RuntimeFilter runtimeFilter) {
      * 2. Only olap scan nodes are supported:
      */
     private void assignRuntimeFilters(ScanNode scanNode) {
-        if (!(scanNode instanceof OlapScanNode)) {
+        if (!(scanNode instanceof OlapScanNode) && !(scanNode instanceof ExternalFileScanNode)) {
```
Comment (Jibing-Li) [inline fe/fe-core/src/main/java/org/apache/doris/planner/RuntimeFilterGenerator.java:325]: I suppose ExternalFileScanNode is used only when `enable_multi_catalog` is set to true.

**Review by Jibing-Li — DISMISSED** [review]
LGTM

File: fe/fe-core/src/main/java/org/apache/doris/planner/RuntimeFilterGenerator.java (hunk 2 above)
Comment (morningman) [reply fe/fe-core/src/main/java/org/apache/doris/planner/RuntimeFilterGenerator.java:325]: Yes

File: be/src/vec/exec/file_scan_node.cpp (hunk 1 above)
Comment (morningman) [reply be/src/vec/exec/file_scan_node.cpp:132]: No, this only take effect when compiling BE with DEBUG mode

**Review by yiguolei — APPROVED** [review]
lgtm

**Pull request merged**